app/views/membership_applications/_reason_waiting.html.haml

```diff
@@ -28,13 +28,38 @@
                               method: 'put' } }
 
 
-
 :javascript
 
-  $('#member_app_waiting_reasons').on('ajax:success', function (e, data) {
-    if (data === "#{@other_waiting_reason_value}") {
-      $('#other-text-field').show();
-    } else {
-      $('#other-text-field').hide();
-    }
+  var ChangeStatus = {
+    unsaved_changes: false
+  }
+
+  $(document).ready(function () {
+
+    $('#member_app_waiting_reasons').on('ajax:success', function (e, data) {
+      if (data === "#{@other_waiting_reason_value}") {
+        $('#other-text-field').show();
+      } else {
+        $('#other-text-field').hide();
+      }
+    });
+
+    $('#custom_reason_text').on('input', function (e) {
+      ChangeStatus.unsaved_changes = true;
+    });
+
+    $('#custom_reason_text').on('change', function (e) {
+      ChangeStatus.unsaved_changes = false;
+    });
+
+    $(window).on('beforeunload', function (e) {
+
+      if ($('#member_app_waiting_reasons').val() === '#{@other_waiting_reason_value}' &&
+          ChangeStatus.unsaved_changes === true) {
+
+        e.returnValue = 'Unsaved changes! Do you want to leave the page?';
```

Comment:
Subsequent to this, I had to introduce Selenium/Chrome as a cucumber test option in order to get the added scenario (for this feature) to run.  So, I changed to `let` and also changed 3 existing scenarios to use selenium.